### PR TITLE
Add deadlock device

### PR DESCRIPTION
**Pull request by Nicogene**
This PR add a `BROKEN_TEST` that reproduces the close deadlock of a device after a `prepare()` on a closed port (#1236).

Please review code.

### CODE REVIEW

**Comment by traversaro** [comment]
I think it would be good to have a reference to the bug/problem that this broken test is exposing,

**Comment by traversaro** [comment]
Thanks @Nicogene !

**Comment by Nicogene** [comment]
You are welcome ;)

File: tests/libYARP_OS/PortTest.cpp [hunk 1]
```diff
@@ -28,14 +29,73 @@
...
+    }
+}
+
+/**
+ * @ingroup dev_impl_media
+ *
+ * A fake camera for testing.
+ */
+class yarp::dev::BrokenDevice : public DeviceDriver,
+                                    public yarp::os::RateThread
```
Comment (drdanz) [inline tests/libYARP_OS/PortTest.cpp]: Nitpick: check the indentation here

File: tests/libYARP_OS/PortTest.cpp [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #include <yarp/os/impl/Logger.h>
 #include <yarp/os/NetType.h>
 #include <yarp/os/impl/UnitTest.h>
+#include <yarp/os/LogStream.h>
```
Comment (drdanz) [inline tests/libYARP_OS/PortTest.cpp]: I don't see this used, please remove it, I don't think it's a good idea to use logstream in Port tests...

**Pull request merged**